Library/Homebrew/download_strategy.rb

```diff
@@ -483,7 +483,13 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
 
       if (filename_with_encoding = content_disposition.parameters["filename*"])
         encoding, encoded_filename = filename_with_encoding.split("''", 2)
-        filename = URI.decode_www_form_component(encoded_filename).encode(encoding) if encoding && encoded_filename
+        # If the `filename*` has incorrectly added double quotes, e.g.
+        #   content-disposition: attachment; filename="myapp-1.2.3.pkg"; filename*=UTF-8''"myapp-1.2.3.pkg"
+        # Then the encoded_filename will come back as the empty string, in which case we should fall back to the
+        # `filename` parameter.
```

Comment:
I don't quite understand.

Wouldn't your example yield `encoded_filename` being `"\"myapp-1.2.3.pkg\""` instead of an empty string?

Reply:
I would have expected something like that too, but no:

```ruby
❯ brew irb --pry
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
[1] brew(main)> line = 'content-disposition: attachment; filename="myapp-1.2.3.pkg"; filename*=UTF-8''"myapp-1.2.3.pkg"'
=> "content-disposition: attachment; filename=\"myapp-1.2.3.pkg\"; filename*=UTF-8\"myapp-1.2.3.pkg\""
[2] brew(main)> content_disposition = Mechanize::HTTP::ContentDispositionParser.new.parse(line.sub(/; *$/, ""), true)
=> #<struct Mechanize::HTTP::ContentDisposition type="attachment", filename="myapp-1.2.3.pkg", creation_date=nil, modification_date=nil, read_date=nil, size=nil, parameters={"filename*"=>"UTF-8"}>
```